### PR TITLE
Jpennors/deployment process

### PR DESCRIPTION
**Pull request by jpennors**
Create script to automate deployment on TMW FTP Server

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        if not any(file_path.startswith(file) for file in FORBIDDEN_DIRECTORIES) and not any(file == file_to_ignore for file_to_ignore in SERVER_FILES_TO_IGNORE):
+            if os.path.isfile(file_path):
+                ftp.delete(file_path)
+            elif os.path.isdir(file_path):
+                remove_files(file_path)
+
+
+def upload_file(path, file):
+    file_path = f'{path}/{file}'
+    ftp.storbinary('STOR ' + path + '/' + file, open(file_path, 'rb'))
```
Comment (abrasseu) [inline deploy.py]: I would rather use this to make sure the file is closed after use:
```python
with open(file_path, 'rb') as fs:
    ftp.storbinary('STOR ' + file_path, fs)
```

File: deploy.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    for file in files:
+        file_path = f'{path}/{file}'
+        if os.path.isfile(file_path):
+            upload_file(path, file)
+        elif os.path.isdir(file_path):
+            if not any(file_path.startswith(dir) for dir in FORBIDDEN_DIRECTORIES):
+                existing_directory = False
+                if file in server_files:
+                    existing_directory = True
+                upload_dir(file_path, existing_directory)
```
Comment (abrasseu) [inline deploy.py]: Simpler
```python
upload_dir(file_path, existing_directory=(file in server_files))
```

File: deploy.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+files_to_ignore = [
+    './public/storage',
+    './public/css/semantic',
+]
+
+def remove_files(path):
+    files = ftp.nlst(path)
+    for file in files:
+        file_path = f'{path}/{file}'
+        if not any(file_path.startswith(file) for file in FORBIDDEN_DIRECTORIES) and not any(file == file_to_ignore for file_to_ignore in SERVER_FILES_TO_IGNORE):
```
Comment (abrasseu) [inline deploy.py]: Shorter like this
```python
if not any(map(file_path.startswith, FORBIDDEN_DIRECTORIES)) and not file in SERVER_FILES_TO_IGNORE:
```

File: deploy.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+import requests
+import configparser
+from ftplib import FTP, FTP_TLS
+
+
+class bcolors:
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
```
Comment (abrasseu) [inline deploy.py:14]: You can do `print_blue`, `print_green`, etc... functions for easier use. For example:
```python
from functools import partiel
def printc(color: str, text: str):
    print(getattr(bcolors, color) + text + colors.ENDC
print_blue = partial(printc, color='OKBLUE')
```

**Review by abrasseu — APPROVED** [review]
Noice

**Review by lancelotimb — APPROVED** [review]
Looks good to me for the logic! Well done 💯💯💯

**Pull request merged**